gitpod-io/gitpod
# GetUpcomingInvoice for BillingService

## Description
This PR implements ~`GetLatestInvoice`~  `GetUpcomingInvoice`. The rename is done intentionally, to a) align with terminology used in stripe, and b) to distinguish between the next invoice and an already paid invoice. The upcoming invoice contains the the amount of credits currently used (🤞🏻 after discounts), which can be compared with the spending limit configured.

- [x] rename GetLatestInvoice to GetUpcomingInvoice

## Related Issue(s)

Fixes #11692 

## How to test
Setup UBB in for a team and lower the spending limit to 1 or 2 credits for a simple test, then start a workspace or two to hit that limit. After a while, cannot tell it precisely, you'll need to reload the dashboard to refresh the spending limit checks. In the `server` component you'll find a log message on spending limit events, and the known dashboard notification should be visible:
<img width="1417" alt="Screen Shot 2022-08-23 at 16 21 48" src="https://user-images.githubusercontent.com/914497/186183902-b7212eff-55f6-41f2-ba43-ea95cdc654b4.png">


## Release Notes

```release-note
NONE
```



## Werft options:
